pimd/pim_iface.c

```diff
@@ -979,14 +979,14 @@ int pim_if_add_vif(struct interface *ifp, bool ispimreg, bool is_vxlan_term)
 	}
 
 	if (ifp->ifindex < 0) {
-		zlog_warn("%s: ifindex=%d < 1 on interface %s", __func__,
+		zlog_warn("%s: ifindex=%d < 0 on interface %s", __func__,
 			  ifp->ifindex, ifp->name);
 		return -2;
-	} else if ((ifp->ifindex == 0) &&
+	} else if ((ifp->ifindex == PIM_OIF_PIM_REGISTER_VIF) &&
 		   ((strncmp(ifp->name, "pimreg", 6)) &&
 		    (strncmp(ifp->name, "pim6reg", 7)))) {
-		zlog_warn("%s: ifindex=%d == 0 on interface %s", __func__,
-			  ifp->ifindex, ifp->name);
+		zlog_warn("%s: ifindex=%d == %d on interface %s", __func__,
```

Comment:
why do we need to print both the values here? This will hit only when its same, so let's keep the logs simpler.
ifindex=%d == %d

Reply:
Done. Please check it right or not, thanks.